Sming/SmingCore/Network/Http/HttpCommon.cpp

```diff
@@ -0,0 +1,69 @@
+/****
+ * Sming Framework Project - Open Source framework for high efficiency native ESP8266 development.
+ * Created 2015 by Skurydin Alexey
+ * http://github.com/SmingHub/Sming
+ * All files of the Sming Core are provided under the LGPL v3 license.
+ *
+ * @author: 2018 - Mikee47 <mike@sillyhouse.net>
+ *
+ * 	httpGetErrnoName() and httpGetStatusText() functions added
+ *
+ ****/
+
+#include "HttpCommon.h"
+
+// Define flash strings and lookup table for HTTP error names
+#define XX(_n, _s) static DEFINE_FSTR(hpename_##_n, "HPE_" #_n);
+HTTP_ERRNO_MAP(XX)
+#undef XX
+
+static FSTR_TABLE(hpeNames) = {
+#define XX(_n, _s) FSTR_PTR(hpename_##_n),
+	HTTP_ERRNO_MAP(XX)
+#undef XX
+};
+
+String httpGetErrnoName(enum http_errno err)

```

Comment:
Needs further discussion: One other thing that is coming to my mind is that maybe we should think about removing the `http` prefix from those functions and replace it with a C++ namespace.

Reply:
I've a similar treatment for WebConstants.h, which lives in the ContentType namespace.
That also involves a change from const char* to String return value, which raises the same issue of compatibility.

Reply:
How about ```namespace HttpText``` ?

Reply:
OK